File: C/countSumOfTwoRepresentations3.c
// int countSumOfTwoRepresentations3(int n, int l, int r) {
//     return fmax(n / 2 - fmax(l, n-r) + 1, 0);
// }

#define countSumOfTwoRepresentations3(n, l, r) fmax(n / 2 - fmax(l, n-r) + 1, 0)

File: Rust/countSumOfTwoRepresentations3.rs
// use std::cmp::max;

fn countSumOfTwoRepresentations3(n: i32, l: i32, r: i32) -> i32 {
    // max(n / 2 - max(l, n-r) + 1, 0)
    0.max(n / 2 - l.max(n-r) + 1)
}

